### PR TITLE
feat: add custom where clause in bulk_update_models and bulk_upsert_models

### PR DESCRIPTION
**Pull request by taylor-cedar**
## Summary
Ticket: https://cedar-jira.atlassian.net/browse/CED-67122
This adds the ability to specify a custom `WHERE` clause when updating models in bulk. This is necessary to check revision numbers or last_updated dates before updating 

##### How can reviewers test your change manually?
Check the new tests. You can use the new param called `update_where`
##### What automated tests did you add? If none, please state why.
Yes, updated tests for new argument and all existing tests still pass

## Security Impact
Please answer the questions below about your PR. 
If yes, please explain and add `cedar-team/security` as reviewers.

##### Add/modify authentication, authorization, encryption, or HTTP headers?
No
##### Add/modify any PII or PHI data fields?
No
##### Add new third parties the app interacts with (e.g., Nordis/Stripe)?
No
##### Ask patients to enter a new kind of data, (e.g. insurance capture)?
No
##### Impact any of the other [Security Engineering Practices](https://careportal.atlassian.net/wiki/spaces/SEC/pages/1192690295/Security+Engineering+Practices)?
_Protect sensitive data, Validate your inputs, Know your user, Establish trust boundaries, Keep good records, Expect vulnerabilities, Follow least privilege_

No

### CODE REVIEW

File: tests/test_bulk_upsert_models.py [hunk 1]
```diff
@@ -551,3 +551,46 @@ def test_complex_update_if_null(self):
...
+        bulk_upsert_models(
+            [model1, model2],
+            update_where=update_where
+        )
+
+        # First model should be updated because 5 > 1
+        saved_model1 = TestComplexModel.objects.get(integer_field=5)
+        self.assertEqual(saved_model1.string_field, "b")
+
+        # Second model should not be updated because 2 <  3
```
Comment (cpaynerogers) [inline tests/test_bulk_upsert_models.py:594]:
```suggestion
        # Second model should not be updated because 2 < 3
```

File: django_bulk_load/bulk_load.py [hunk 2]
```diff
@@ -219,6 +220,8 @@ def bulk_update_models(
     list is changed) (i.e. update_on/last_modified)
     :param update_if_null_field_names: Fields that only get updated if the new value is NULL or existing
     value in the DB is NULL.
+    :param update_where: Function that returns a Composable that is used to filter the update query. Cannot be used
+    with model_changed_field_names or update_if_null_field_names
```
Comment (cpaynerogers) [inline django_bulk_load/bulk_load.py:224]: I updated (locally) the test you added for `update_where` by adding the following to the call to `bulk_update_models`:

- `update_if_null_field_names=["string_field"]`

And this causes the test to fail (this seems expected). But I added many variations of `model_changed_field_names` and things seem to work as intended in the simple test case. Just flagging that it might make sense to clarify this docstring, or if we're worried about unexpected behavior, should we sanitize the inputs to `bulk_update_models` and `bulk_upsert_models` to check that callers aren't using `update_where` and these alternatives and either warn or (my preference) fail to run the load operation.

```suggestion
    :param update_where: Function that returns a Composable that is used to filter the update query. Should not be used
    with model_changed_field_names or update_if_null_field_names (can lead to unexpected behavior)
```

**Review by cpaynerogers — DISMISSED** [review]
These changes make sense to me, just a few minor, non-blocking comments.

- [x] I fully understand what the code is doing and its context
- [x] The code is the best way to accomplish this problem (I would code it similarly)
- [x] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [ ] The code handles edge cases (that I can think of) and has tests to validate them
- [x] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)

**Review by cpaynerogers — APPROVED** [review]
- [x] I fully understand what the code is doing and its context
- [x] The code is the best way to accomplish this problem (I would code it similarly)
- [x] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [x] The code handles edge cases (that I can think of) and has tests to validate them
- [x] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)
